File: android/src/main/java/org/medlix/medlix_data_vault/FlutterSecureStorage.java
package org.medlix.medlix_data_vault;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.security.crypto.EncryptedSharedPreferences;
import androidx.security.crypto.MasterKeys;

import java.io.IOException;
import java.security.GeneralSecurityException;

public class FlutterSecureStorage {
    private static final String PREFERENCES_FILE_NAME = "encrypted_preferences";

    private SharedPreferences preferences;

    public FlutterSecureStorage(Context context) {
        try {
            String masterKeyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC);
            preferences = EncryptedSharedPreferences.create(
                    PREFERENCES_FILE_NAME,
                    masterKeyAlias,
                    context,
                    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
            );
        } catch (GeneralSecurityException | IOException e) {
            // Handle error
            e.printStackTrace();
        }
    }

    public String read(String key) {
        return preferences.getString(key, null);
    }

    public void write(String key, String value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public void delete(String key) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.remove(key);
        editor.apply();
    }
}

File: android/src/main/kotlin/org/medlix/medlix_data_vault/FlutterSecureStorage.kt
package org.medlix.medlix_data_vault

import android.content.Context
import android.content.SharedPreferences
import android.util.Log
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKeys
import java.io.IOException
import java.security.GeneralSecurityException

class FlutterSecureStorage(context: Context) {
    private lateinit var preferences: SharedPreferences

    init {
        try {
            val masterKeyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC)

            preferences = EncryptedSharedPreferences.create(
                PREFERENCES_FILE_NAME,
                masterKeyAlias,
                context,
                EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
            )
        } catch (e: GeneralSecurityException) {
            // Handle error
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }

    fun read(key: String?): String? {
        return preferences.getString(key, null)
    }

    fun readAll(): Map<String, String> {
        return preferences.all.mapValues { it.value.toString() }
    }

    fun write(key: String?, value: String?): Boolean {
        val editor = preferences.edit()
        editor.putString(key, value)
        editor.apply()
        return preferences.contains(key)
    }

    fun delete(key: String?): Boolean {
        val editor = preferences.edit()
        return editor.remove(key).commit()
    }

    fun deleteAll(): Int {
        val size = preferences.all.size
        val editor = preferences.edit()
        return if (editor.clear().commit()) {
            size
        } else {
            0
        }
    }

    companion object {
        private const val PREFERENCES_FILE_NAME = "encrypted_preferences"
    }
}
